app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PiecesController < ApplicationController
+  def new
+    set_white_pieces
+    set_black_pieces
+    pawn_pos = 1
+    8.times do
+      Piece.create(piece_name: 'pawn_white', x_position: pawn_pos, y_position: 2)
+      Piece.create(piece_name: 'pawn_black', x_position: pawn_pos, y_position: 7)
+      pawn_pos += 1
```

Comment:
This seems unnecessary. You should be able to obtain the loop iteration count from the loop, then use it +1 for the x_pos